From: Caleb Connolly <caleb@connolly.tech>
Date: Sun, 20 Mar 2022 21:24:28 +0000
Subject: arm64: dts: rk3399-pinephone-pro: add pinconf for stk3311

Add a pinconf for the stk3311 to configure it's interrupt GPIO as an
input. The BROM configures it as output by default.

Signed-off-by: Caleb Connolly <caleb@connolly.tech>
---
 arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts b/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts
index 1ca8b5f..f945d89 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts
@@ -976,6 +976,8 @@ light-sensor@48 {
 		interrupt-parent = <&gpio4>;
 		interrupts = <RK_PD3 IRQ_TYPE_EDGE_FALLING>;
 		proximity-near-level = <32>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&stk3311_int>;
 		vdd-supply = <&vcc_3v0>;
 		leda-supply = <&vcc_3v0>;
 	};
@@ -1338,6 +1340,12 @@ spk_en: spk-en {
 			rockchip,pins = <0 RK_PB3 RK_FUNC_GPIO &pcfg_pull_none>;
 		};
 	};
+
+	lights {
+		stk3311_int: stk3311-int {
+			rockchip,pins = <4 RK_PD3 RK_FUNC_GPIO &pcfg_input_pull_up>;
+		};
+	};
 };
 
 &pwm0 {
